h2o-core/src/main/java/water/H2O.java

```diff
@@ -2132,28 +2131,30 @@ public static boolean isNodeInFlatfile(H2ONode node) {
   }
 
   /**
-   * Is manual multicast enabled?
+   * Check if manual multi-cast is enabled.
    * @return  true if `-flatfile` option was specified on commandline
    */
   public static boolean isFlatfileEnabled() {
     return STATIC_H2OS != null;
   }
 
   /** Setup a set of nodes which should be contacted during
-   * manual multicast.
-   * @param nodes  set of H2O nodes.
+   * @param nodes set of H2O nodes
    */
-  public static void setFlatfile(HashSet<H2ONode> nodes) {
-    STATIC_H2OS = nodes;
+  public static void setFlatfile(Set<H2ONode> nodes) {
+    if(nodes == null){
+      STATIC_H2OS = null;
+    }else{
+      STATIC_H2OS = Collections.newSetFromMap(new ConcurrentHashMap<H2ONode, Boolean>());
```

Comment:
Make it concurrent-safe

Reply:
Just to make sure: does a call to `setFlatFile` need to be thread-safe to?